agent/agent.go

```diff
@@ -246,7 +246,7 @@ func (a *Agent) TempDir() string {
 }
 
 // CreateTemp Creates a temporary directory so that we may gather results and files before compressing the final
-//  artifact.
+// artifact.
```

Comment:
Removing these spaces prevents `go fmt` from going wild and adding an empty comment line when I save -- hope that's ok!

Reply:
Yeah, where there are extra spaces in doc comments, we should remove them. Go Doc interprets these as pre-formatted text (code blocks). See the comment for agent.CreateTemp() at https://pkg.go.dev/github.com/hashicorp/hcdiag@v0.4.0/agent#Agent.CreateTemp
<img width="1196" alt="image" src="https://user-images.githubusercontent.com/16619375/193862711-3d9adb93-6ef3-4f99-9312-8877620f6567.png">

Reply:
As a frequent space-placer, I'd been adding the space because it makes goland highlight the next line with the same comment font. Didn't realize it was breaking things downstream. No need to keep the spaces!

Reply:
Just created https://github.com/hashicorp/hcdiag/pull/243 for this